packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/OutputTypes/TypeFactory.cs

```diff
@@ -69,13 +68,13 @@ public abstract class TypeFactory
         public abstract CSharpType PageResponseType();
 
         /// <summary>
-        /// The type for change tracking lists.
+        /// The type of the implementation type of list properties. The type should implement both <see cref="IList{T}"/> and <see cref="IReadOnlyList{T}"/>
         /// </summary>
-        public virtual CSharpType ChangeTrackingListType => ChangeTrackingListProvider.Instance.Type;
+        public virtual CSharpType ListImplementationType => ChangeTrackingListProvider.Instance.Type;
 
         /// <summary>
-        /// The type for change tracking dictionaries.
+        /// The type of the implementation type of dictionary properties. The type should implement both <see cref="IDictionary{TKey, TValue}"/> and <see cref="IReadOnlyDictionary{TKey, TValue}"/>
```

Comment:
Similar comment here:
```suggestion
        /// The type that represents the implementation of dictionary properties. This type should implement both <see cref="IDictionary{TKey, TValue}"/> and <see cref="IReadOnlyDictionary{TKey, TValue}"/>.
```